compiler/src/typed/typemod.re

```diff
@@ -976,15 +976,15 @@ let get_compilation_mode = () => {
 
 let type_implementation = prog => {
   let sourcefile = prog.prog_loc.loc_start.pos_fname;
-  /* TODO: Do we maybe need a fallback here? */
+  // TODO: Do we maybe need a fallback here?
   let modulename =
     Grain_utils.Filepath.String.filename_to_module_name(sourcefile);
   Env.set_unit((modulename, sourcefile, get_compilation_mode()));
   let initenv = initial_env();
   let (stritems, sg, finalenv) = type_module(initenv, prog);
   let (statements, env) = stritems;
   let simple_sg = simplify_signature(sg);
-  let filename = sourcefile; /* TODO: I think this is okay */
+  let filename = sourcefile; // TODO: I think this is okay
```

Comment:
```suggestion
  let filename = sourcefile;
```

Reply:
Btw, I remember this line being an absolute nightmare when I was working on my filepath rework. So I don't think we should remove this comment (because it's only okay *for now*, but not in the long run)

Reply:
Alright. I've always particularly disliked this TODO so this is at least a good opportunity to reword it. Should we make an actual issue?

Reply:
Yeah, though I'd have to dig into my PR to remember what the issue is. If this is the same thing that I'm thinking about, it's that "filename" isn't actually the correct naming and it's more like "relative module identifier" or something? Is this the string that is always a relative path and doesn't include an extension?

Reply:
![image](https://user-images.githubusercontent.com/992373/173494991-f413a609-ef03-46b8-9b81-ad368f704662.png)

Reply:
> Btw, I remember this line being an absolute nightmare when I was working on my filepath rework. So I don't think we should remove this comment (because it's only okay _for now_, but not in the long run)

Ive commited everything but this until an error is up to link for this.

Reply:
@spotandjake You can use https://github.com/grain-lang/grain/issues/1396 for this.